solutions/platinum/cf-1254D.mdx

```diff
@@ -0,0 +1,142 @@
+---
+id: cf-1254D
+source: CF
+title: Tree Queries
+author: Rameez Parwez
+---
+
+## Explanation
+
+We can solve this problem using Heavy-Light Decomposition (HLD) combined with a Fenwick Tree (BIT). The tree is first decomposed into heavy and light edges, and then each node's size and depth are calculated. When updating a node, contributions are added using the BIT, ensuring $\mathcal{O} (\log N)$ complexity per update. To compute the expected value during queries, we aggregate contributions from light vertices and iterate over the heavy vertices.
+
```

Comment:
could you explain some of your intuition for this problem? right now it just feels like a description of the code.